diff --git a/confdb/aclocal_f77.m4 b/confdb/aclocal_f77.m4
index 4504505..f96a0db 100644
--- a/confdb/aclocal_f77.m4
+++ b/confdb/aclocal_f77.m4
@@ -1453,7 +1453,7 @@ if test "X$pac_cv_prog_f77_mismatched_args" = X ; then
 	# The best solution is to turn off errors on particular routines
 	# if that isn't possible (e.g., too many of them), then
 	# just try arguments that turn off all checking
-	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-mismatch" ; do
+	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-w" "-fallow-argument-mismatch" ; do
             testok=no
             FFLAGS="$FFLAGS $flags"
             AC_COMPILE_IFELSE([
diff --git a/src/mpi/romio/confdb/aclocal_f77.m4 b/src/mpi/romio/confdb/aclocal_f77.m4
index 4504505..f96a0db 100644
--- a/src/mpi/romio/confdb/aclocal_f77.m4
+++ b/src/mpi/romio/confdb/aclocal_f77.m4
@@ -1453,7 +1453,7 @@ if test "X$pac_cv_prog_f77_mismatched_args" = X ; then
 	# The best solution is to turn off errors on particular routines
 	# if that isn't possible (e.g., too many of them), then
 	# just try arguments that turn off all checking
-	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-mismatch" ; do
+	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-w" "-fallow-argument-mismatch" ; do
             testok=no
             FFLAGS="$FFLAGS $flags"
             AC_COMPILE_IFELSE([
diff --git a/src/mpl/confdb/aclocal_f77.m4 b/src/mpl/confdb/aclocal_f77.m4
index 4504505..f96a0db 100644
--- a/src/mpl/confdb/aclocal_f77.m4
+++ b/src/mpl/confdb/aclocal_f77.m4
@@ -1453,7 +1453,7 @@ if test "X$pac_cv_prog_f77_mismatched_args" = X ; then
 	# The best solution is to turn off errors on particular routines
 	# if that isn't possible (e.g., too many of them), then
 	# just try arguments that turn off all checking
-	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-mismatch" ; do
+	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-w" "-fallow-argument-mismatch" ; do
             testok=no
             FFLAGS="$FFLAGS $flags"
             AC_COMPILE_IFELSE([
diff --git a/src/pm/hydra/confdb/aclocal_f77.m4 b/src/pm/hydra/confdb/aclocal_f77.m4
index 4504505..f96a0db 100644
--- a/src/pm/hydra/confdb/aclocal_f77.m4
+++ b/src/pm/hydra/confdb/aclocal_f77.m4
@@ -1453,7 +1453,7 @@ if test "X$pac_cv_prog_f77_mismatched_args" = X ; then
 	# The best solution is to turn off errors on particular routines
 	# if that isn't possible (e.g., too many of them), then
 	# just try arguments that turn off all checking
-	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-mismatch" ; do
+	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-w" "-fallow-argument-mismatch" ; do
             testok=no
             FFLAGS="$FFLAGS $flags"
             AC_COMPILE_IFELSE([
diff --git a/src/pm/hydra/mpl/confdb/aclocal_f77.m4 b/src/pm/hydra/mpl/confdb/aclocal_f77.m4
index 4504505..f96a0db 100644
--- a/src/pm/hydra/mpl/confdb/aclocal_f77.m4
+++ b/src/pm/hydra/mpl/confdb/aclocal_f77.m4
@@ -1453,7 +1453,7 @@ if test "X$pac_cv_prog_f77_mismatched_args" = X ; then
 	# The best solution is to turn off errors on particular routines
 	# if that isn't possible (e.g., too many of them), then
 	# just try arguments that turn off all checking
-	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-mismatch" ; do
+	for flags in ifelse($2,yes,,"-wmismatch=foo1") "-w" "-fallow-argument-mismatch" ; do
             testok=no
             FFLAGS="$FFLAGS $flags"
             AC_COMPILE_IFELSE([
